src/utils.js

```diff
@@ -389,3 +394,23 @@ export const safeExecute = (
     logger.warn(message, err);
   }
 };
+
+export const getLumigoEventKey = event => {
+  const noCircularEvent = noCirculars(event);
+  return recursiveGetLumigoEventKey(noCircularEvent);
+};
+
+const recursiveGetLumigoEventKey = noCircularEvent => {
+  let foundValue = undefined;
+  Object.keys(noCircularEvent).some(function(k) {
+    if (k === LUMIGO_EVENT_KEY) {
+      foundValue = noCircularEvent[k];
+      return true;
+    }
+    if (noCircularEvent[k] && typeof noCircularEvent[k] === 'object') {
+      foundValue = recursiveGetLumigoEventKey(noCircularEvent[k]);
+      return foundValue !== undefined;
+    }
+  });
+  return foundValue;
+};
```

Comment:
I think we should create a custom function, and not running the object n*2. the tracer is slow enough

Reply:
Let's talk about it, call me :-)